gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/tpgm/diff/functions/DiffPerElement.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.tpgm.diff.functions;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.java.functions.FunctionAnnotation;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.gradoop.common.model.impl.pojo.temporal.TemporalElement;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.flink.model.api.tpgm.functions.TemporalPredicate;
+import org.gradoop.flink.model.impl.operators.tpgm.diff.Diff;
+
+import java.util.Objects;
+
+/**
+ * Evaluates two temporal predicates per element of a data set and sets the
+ * {@value Diff#PROPERTY_KEY} accordingly.
+ *
+ * @param <E> The element type.
+ */
+@FunctionAnnotation.ReadFields("validTime")
+@FunctionAnnotation.NonForwardedFields("properties")
+public class DiffPerElement<E extends TemporalElement> implements MapFunction<E, E> {
```

Comment:
Here you have to use a flatMap since elements can be discarded if they do not occur in both snapshots